app/controllers/concerns/authenticated_state_file_intake_concern.rb

```diff
@@ -5,13 +5,26 @@ module AuthenticatedStateFileIntakeConcern
     before_action :require_state_file_intake_login
   end
 
+  def current_intake
+    # NOTE: This intake is different from the unauthenticated value, stored in the regular session
+    # (Which the unauthenticated sessions controller uses)
+    unless @intake
+      current_intake_name = "current_#{question_navigator.intake_class.name.underscore}"
+      @intake = self.send(current_intake_name)
+    end
+    @intake
+  end
+
   private
 
   def card_postscript
     I18n.t("state_file.state_file_pages.card_postscript.responses_saved_html").html_safe
   end
 
   def require_state_file_intake_login
+    # NOTE: Most places in the code do not seem to use the authenticated version of
+    # current_state_file_XX_intake - they use current_intake instead which is
+    # unauthenticated.
     if current_state_file_az_intake.blank? && current_state_file_ny_intake.blank?
```

Comment:
This is a good spot to use the current_authenticated_intake. We should be explicit here. and bein authenticated for a different state should not qualify as authenticated for this state.
```suggestion
    if current_authenticated_intake.blank?
```